tiledb/sm/array_schema/array_schema.cc

```diff
@@ -1392,7 +1430,9 @@ ArraySchema ArraySchema::deserialize(
   // Set schema name
   std::string name = uri.last_path_part();
 
-  return ArraySchema(
+  return make_shared<ArraySchema>(
+      HERE(),
+      make_shared<MemoryTracker>(HERE()),
```

Comment:
This will break which means ArraySchema::deserialize needs a new MemoryTracker argument which will likely lead to more changes to get that everywhere it needs to go.

Reply:
Also, you'll probably quickly run into their being a lack of memory tracking available in various API/StorageManager calls. In those cases I'd create a new MemoryTracker::Type for `ARRAY_CREATE` or whatever the case may be.

Reply:
Where should this type be used? I've been creating management types for the member variables instead. Should I just group them under "ARRAY_SCHEMA" or some similar label?

Also, I think what you're suggesting is to move the memory tracker argument to be declared in deserialize instead--is this correct?